lib/css/components/notices.less

```diff
@@ -1,252 +1,267 @@
-.construct-notice-component() {
-    background: var(--_no-bg, var(--black-050));
-    border-color: var(--_no-bc, var(--bc-medium));
-    border-style: solid;
-    color: var(--_no-fc, var(--fc-medium));
-    font-size: var(--fs-body1);
-
-    code {
-        background: var(--_no-code-bg, transparent);
-    }
-
-    & &--btn {
-        color: inherit;
-        padding: var(--su8);
-
-        &:focus,
-        &:hover {
-            background: var(--_no-btn-focus-bg, var(--black-100));
-        }
-
-        &:active {
-            background: var(--_no-btn-active-bg, var(--black-150));
-        }
-    }
+.construct-notice-component(@baseClass) {
+    // BASE COMPONENT-SPECIFIC CUSTOM PROPERTIES
+    --_no-bc: var(--bc-medium);
+    --_no-bg: var(--black-050);
+    --_no-fc: var(--fc-medium);
+    --_no-btn-bg-focus: var(--black-100);
+    --_no-btn-bg-active: var(--black-150);
 
+    // MODIFIERS
     &:not(&__important) {
-        .dark-mode({ --_no-bc: var(--_no-bg); });
-        .highcontrast-mode({ --_no-bc: currentColor; });
-        .highcontrast-dark-mode({ --_no-bc: currentColor; });
+        .dark-mode({
+            --_no-bc: var(--_no-bg);
+        });
+        .highcontrast-mode({
+            --_no-bc: currentColor;
+        });
+        .highcontrast-dark-mode({
+            --_no-bc: currentColor;
+        });
```

Comment:
This component has me reconsidering the `// CONTEXTUAL STYLES` section in many components. It would've been so messy to have it in this component that I just included the contextual styles (high contrast, dark, etc) within the modifier/variants.